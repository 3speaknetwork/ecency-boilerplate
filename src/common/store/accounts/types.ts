import { ExtendedAccount } from "@esteemapp/dhive";

export interface Account extends ExtendedAccount {}

export interface State {
  data: Account;
  loading: boolean;
  error: boolean;
}

export enum ActionTypes {
  FETCH = "@account/FETCH",
  FETCHED = "@account/FETCHED",
  FETCH_ERROR = "@account/FETCH_ERROR"
}

export interface FetchAction {
  type: ActionTypes.FETCH;
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  data: Account;
}

export interface FetchErrorAction {
  type: ActionTypes.FETCH_ERROR;
}

export type Actions = FetchAction | FetchedAction | FetchErrorAction;