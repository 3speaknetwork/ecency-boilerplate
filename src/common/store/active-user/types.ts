import { Account } from "../accounts/types";

export interface ActiveUser {
  name: string;
  data: Account;
}

export enum ActionTypes {
  RELOAD = "@active-user/RELOAD",
  UPDATE = "@active-user/UPDATE",
}

export interface ReloadAction {
  type: ActionTypes.RELOAD;
}

export interface UpdateAction {
  type: ActionTypes.UPDATE;
  data: Account;
}

export type Actions = ReloadAction | UpdateAction;
