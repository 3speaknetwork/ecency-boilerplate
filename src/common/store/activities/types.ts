interface BaseActivity {
    event: "notify";
    source: string;
    target: string;
    timestamp: string;
    extra: any;
}

export interface VoteActivity extends BaseActivity {
    type: "vote",
    extra: {
        permlink: string;
        weight: number;
        title: string | null;
        img_url: string | null;
    }
}

export interface MentionActivity extends BaseActivity {
    type: "mention",
    extra: {
        permlink: string;
        is_post: 0 | 1;
        title: string | null;
        img_url: string | null;
    }
}

export interface FollowActivity extends BaseActivity {
    type: "follow",
    extra: {
        what: string[]
    }
}

export interface ReplyActivity extends BaseActivity {
    type: "reply",
    extra: {
        title: string;
        body: string;
        json_metadata: string;
        permlink: string;
        parent_author: string;
        parent_permlink: string;
        parent_title: string | null;
        parent_img_url: string | null;
    }
}

export interface ReblogActivity extends BaseActivity {
    type: "reblog";
    extra: {
        permlink: string;
        title: string | null;
        img_url: string | null;
    }

}

export interface TransferActivity extends BaseActivity {
    type: "transfer";
    extra: {
        amount: string;
        memo: string;
    }
}


export type Activity = VoteActivity | MentionActivity | FollowActivity | ReplyActivity | ReblogActivity | TransferActivity;
