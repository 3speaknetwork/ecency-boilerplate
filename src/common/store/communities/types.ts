export interface Community {
    about: string,
    avatar_url: string,
    created_at: string,
    description: string,
    flag_text: string,
    id: number,
    is_nsfw: boolean,
    lang: string,
    name: string,
    num_authors: number,
    num_pending: number,
    subscribers: number,
    sum_pending: number,
    team: Array<Array<string>>,
    title: string,
    type_id: number
}

export interface State {
    list: Record<string, Community>,
}

export enum ActionTypes {
    FETCHED = '@communities/FETCHED',
}


export interface FetchedAction {
    type: ActionTypes.FETCHED;
    name: string,
    data: Community
}

export type Actions = FetchedAction; // | .. | ..
