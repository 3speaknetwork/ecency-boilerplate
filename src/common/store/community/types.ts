export interface Community {
  about: string;
  avatar_url: string;
  created_at: string;
  description: string;
  flag_text: string;
  id: number;
  is_nsfw: boolean;
  lang: string;
  name: string;
  num_authors: number;
  num_pending: number;
  subscribers: number;
  sum_pending: number;
  settings?: any;
  team: Array<Array<string>>;
  title: string;
  type_id: number;
}

export interface State extends Community {}

export enum ActionTypes {
  FETCHED = "@communities/FETCHED",
  RESET = "@communities/RESET",
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  data: Community;
}

export interface ResetAction {
  type: ActionTypes.RESET;
}

export type Actions = FetchedAction | ResetAction;
