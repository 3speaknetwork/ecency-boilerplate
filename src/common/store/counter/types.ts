export interface State {
    val: number
}

export enum ActionTypes {
    SET_COUNTER = '@counter/SET_COUNTER',
    INCREMENT_COUNTER = '@counter/INCREMENT_COUNTER',
    DECREMENT_COUNTER = '@counter/DECREMENT_COUNTER',
}

export interface SetAction {
    type: ActionTypes.SET_COUNTER;
    value: string;
}

export interface IncrementAction {
    type: ActionTypes.INCREMENT_COUNTER;
}

export interface DecrementAction {
    type: ActionTypes.DECREMENT_COUNTER;
}

export type Actions = SetAction | IncrementAction | DecrementAction
