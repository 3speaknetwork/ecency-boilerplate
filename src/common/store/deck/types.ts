export interface DeckModel {
  data?: any;
  dataParams: any[];
  header: {
    title: string;
    icon: JSX.Element | null;
    reloading: boolean;
  };
  listItemComponent: any;
  createdAt: Date;
  // TODO: Add owner
}

export interface IdentifiableDeckModel extends DeckModel {
  id: string;
  content: string;
}

export enum ActionTypes {
  CREATE = "@deck/CREATE",
  SET_DATA =  "@deck/FETCH_DATA",
  SET_RELOADING = "@deck/SET_RELOADING",
  REORDER = "@deck/REORDER"
}

export interface CreateAction {
  type: ActionTypes.CREATE;
  data: [
    DeckModel['listItemComponent'],
    DeckModel['header']['title'],
    DeckModel['header']['icon'],
    DeckModel['dataParams']
  ];
}

export interface SetDataAction {
  type: ActionTypes.SET_DATA;
  data: {
    title: string;
    data: any;
  };
}

export interface SetReloadingAction {
  type: ActionTypes.SET_RELOADING;
  data: {
    title: string;
    isReloading: boolean;
  };
}

export interface ReOrderAction {
  type: ActionTypes.REORDER;
  data: {
    startIndex: number;
    endIndex: number;
  }
}

export type DeckState = {
  items: IdentifiableDeckModel[];
  isContentLoading: boolean;
};
export type Actions = CreateAction | SetDataAction | SetReloadingAction;