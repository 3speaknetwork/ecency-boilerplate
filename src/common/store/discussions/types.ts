import {Entry} from "../entries/types";

export interface Discussions {
    list: Record<string, Entry>;
    loading: boolean;
    error: boolean;
}

export enum ActionTypes {
    FETCH = "@discussions/FETCH",
    FETCHED = "@discussions/FETCHED",
    FETCH_ERROR = "@discussions/FETCH_ERROR",
    RESET = "@discussions/RESET",
}

export interface FetchAction {
    type: ActionTypes.FETCH;
}

export interface FetchedAction {
    type: ActionTypes.FETCHED;
    list: Record<string, Entry>;
}

export interface FetchErrorAction {
    type: ActionTypes.FETCH_ERROR;
}

export interface ResetAction {
    type: ActionTypes.RESET;
}

export type Actions = FetchAction | FetchedAction | FetchErrorAction | ResetAction;
