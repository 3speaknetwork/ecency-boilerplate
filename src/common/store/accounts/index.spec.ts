import reducer, { initialState, addAct } from "./index";

import { State } from "./types";

let state: State = [];

it("1- default state", () => {
  expect(state).toMatchSnapshot();
});

it("2- addAct - should add account", () => {
  state = reducer(state, addAct({ name: "user1" }));
  expect(state).toMatchSnapshot();
});

it("3- addAct - should add account", () => {
  state = reducer(state, addAct({ name: "user2" }));
  expect(state).toMatchSnapshot();
});

it("4- addAct - should update account", () => {
  state = reducer(state, addAct({ name: "user1", post_count: 10, __loaded: true }));
  expect(state).toMatchSnapshot();
});
