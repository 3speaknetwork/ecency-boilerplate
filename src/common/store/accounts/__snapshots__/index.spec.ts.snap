// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `Array []`;

exports[`2- addAct - should add account 1`] = `
Array [
  Object {
    "name": "user1",
  },
]
`;

exports[`3- addAct - should add account 1`] = `
Array [
  Object {
    "name": "user1",
  },
  Object {
    "name": "user2",
  },
]
`;

exports[`4- addAct - should update account 1`] = `
Array [
  Object {
    "name": "user2",
  },
  Object {
    "__loaded": true,
    "active": Object {
      "account_auths": Array [],
      "key_auths": Array [
        Array [
          "STM71z4rmzGHdp7pmePZyS1G2GrbuBqg9PPHRgiK6uYqjzUTRNCRR",
          1,
        ],
      ],
      "weight_threshold": 1,
    },
    "balance": "1.751 HIVE",
    "created": "2018-01-29T20:10:06",
    "delegated_vesting_shares": "145395.758709 VESTS",
    "follow_stats": Object {
      "account": "talhasch",
      "follower_count": 232,
      "following_count": 27,
    },
    "hbd_balance": "0.311 HBD",
    "json_metadata": "{\\"profile\\":{\\"name\\":\\"Talha B.\\",\\"about\\":\\"Founder @runkod, Software Developer @esteemapp, Lifelong Learner\\",\\"cover_image\\":\\"https://img.esteem.app/rwd380.jpg\\",\\"profile_image\\":\\"https://img.esteem.app/821e0q.jpg\\",\\"website\\":\\"https://github.com/talhasch\\",\\"location\\":\\"Istanbul\\"},\\"escrow\\":{\\"fees\\":{\\"STEEM\\":\\"1.000\\",\\"SBD\\":\\"3.000\\"},\\"terms\\":\\"lorem ipsum dolor sit amet\\"}}",
    "memo_key": "STM7rioGL7NopT2Zo446hnzAiHLp5sJ1gS55xu3NikmxfQnEznsue",
    "name": "user1",
    "next_vesting_withdrawal": "1969-12-31T23:59:59",
    "post_count": 261,
    "posting": Object {
      "account_auths": Array [
        Array [
          "ecency.app",
          1,
        ],
        Array [
          "esteem-app",
          1,
        ],
        Array [
          "esteem.app",
          1,
        ],
        Array [
          "esteemapp",
          1,
        ],
        Array [
          "peakd.app",
          1,
        ],
      ],
      "key_auths": Array [
        Array [
          "STM6uvU7j624wCZNa2pcXvkqmbnNC1cgraDrVG3pFRhXvj9LYS7Xp",
          1,
        ],
      ],
      "weight_threshold": 1,
    },
    "posting_json_metadata": "{\\"profile\\":{\\"name\\":\\"Talha B.\\",\\"about\\":\\"Founder @runkod, Software Developer @ecency, Lifelong Learner\\",\\"cover_image\\":\\"https://images.ecency.com/DQmTsFJJV7gXpsHNatEWz1vgzykaM4WByxAgLMzWshmXNrj/cover.jpg\\",\\"profile_image\\":\\"https://images.ecency.com/DQmV2J1oAUoKxjyJQTu3TcZfGxDr5fzajMGhHEd1e1GALca/ben.jpg\\",\\"website\\":\\"https://github.com/talhasch\\",\\"location\\":\\"Istanbul\\",\\"version\\":2}}",
    "profile": Object {
      "about": "Founder @runkod, Software Developer @ecency, Lifelong Learner",
      "cover_image": "https://images.ecency.com/DQmTsFJJV7gXpsHNatEWz1vgzykaM4WByxAgLMzWshmXNrj/cover.jpg",
      "location": "Istanbul",
      "name": "Talha B.",
      "profile_image": "https://images.ecency.com/DQmV2J1oAUoKxjyJQTu3TcZfGxDr5fzajMGhHEd1e1GALca/ben.jpg",
      "website": "https://github.com/talhasch",
    },
    "proxied_vsf_votes": Array [
      0,
      0,
      0,
      0,
    ],
    "proxy": "",
    "received_vesting_shares": "0.000000 VESTS",
    "reputation": "1021265023022",
    "reward_hbd_balance": "0.000 HBD",
    "reward_hive_balance": "0.000 HIVE",
    "reward_sbd_balance": "0.000 HBD",
    "reward_vesting_balance": "0.000000 VESTS",
    "reward_vesting_hive": "0.000 HIVE",
    "savings_balance": "1.000 HIVE",
    "savings_hbd_balance": "0.002 HBD",
    "savings_sbd_balance": "0.002 HBD",
    "to_withdraw": "87017000000",
    "vesting_shares": "151590.952150 VESTS",
    "vesting_withdraw_rate": "0.000000 VESTS",
    "voting_manabar": Object {
      "current_mana": "6195193441",
      "last_update_time": 1607690265,
    },
    "withdrawn": "87017000000",
    "witness_votes": Array [
      "blocktrades",
      "good-karma",
      "gtg",
    ],
  },
]
`;
